src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -2512,3 +2496,76 @@ export function updateSavedUserPreferenceChartIds(
     }
     return chartSettings;
 }
+
+export function getMolecularProfileOptions(
+    molecularProfiles: MolecularProfile[],
+    molecularPorfileSampleSet: { [id: string]: string[] }
+): GenomicDataCountWithSampleUniqueKeys[] {
+    return _.chain(molecularProfiles)
+        .groupBy(molecularProfile =>
+            molecularProfile.molecularProfileId.replace(
+                molecularProfile.studyId + '_',
+                ''
+            )
+        )
+        .map((profiles, value) => {
+            const uniqueProfiledsamples = _.chain(profiles)
+                .flatMap(
+                    molecularProfile =>
+                        molecularPorfileSampleSet[
+                            molecularProfile.molecularProfileId
+                        ] || []
+                )
+                .uniq()
+                .value();
+            return {
+                value: value,
+                label: profiles[0].name,
+                count: uniqueProfiledsamples.length,
+                sampleUniqueKeys: uniqueProfiledsamples,
+            };
+        })
+        .value();
+}
+
+export function getCNASamplesCount(molecularPorfileSampleCountSet: {
+    [id: string]: number;
+}) {
+    let count = 0;
+    if (molecularPorfileSampleCountSet['cna'] !== undefined) {
```

Comment:
where do these keys come from?  can we make enum for them?  they are sensitive because would result in wrong counts.   also we could do this tallying programmatically